telluric/product_view.py

```diff
@@ -0,0 +1,249 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+from affine import Affine
+from telluric.constants import WGS84_CRS
+from telluric.util.raster_utils import _join_masks_from_masked_array
+
+from .base_renderer import BaseFactory
+
+
+class ProductView():
+    """ ProductView is an abstract class.
+
+    To implement a view class you should inherit from one of its subclasses, ColormapView or BandsComposer.
+    You should set the following attributes if it is not set in the parent class:
+        name display_name description type output_bands required_bands
+    """
+
+    type = np.uint8
+
+    @classmethod
+    def get_name(cls):
+        return cls.name
+
+    @classmethod
+    def to_dict(cls):
+        attributes = "name display_name description type output_bands required_bands".split()
+        d = {}
+        for attr in attributes:
+            d[attr] = getattr(cls, attr)
+        return d
+
+    @classmethod
+    def legend_thumbnail_raster(cls, min, max, band_names, dtype):
```

Comment:
for ppl that are not in the context of telluric explorer we need to explain what is a thubmnail raster.

Somthing like, a raster that can show the visualization that is going to be used, 

maybe this code should be move to one of our internal librararies

Reply:
yes, we need to rethink it

Reply:
I will remove all the thumbnail out of the lib